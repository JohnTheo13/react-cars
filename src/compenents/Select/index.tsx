/**
 * Generic Selec input only accepts array
 * Reusability depends on the expected data structure
 */
import { ReactElement, ChangeEventHandler } from 'react'
import StyledSelect from './style'

interface selectType {
  options: Array<string | number>
  onChange: ChangeEventHandler
  placeholder: string
  name: string
}

const SelectInput = ({
  options,
  onChange,
  placeholder = '',
  name
}: selectType): ReactElement => (
  <StyledSelect data-testid='select' id={name} onChange={onChange} name={name}>
    {!!placeholder && <option value={0}>{placeholder}</option>}
    {options.map(option => (
      <option key={option} value={option}>
        {option}
      </option>
    ))}
  </StyledSelect>
)

export default SelectInput
