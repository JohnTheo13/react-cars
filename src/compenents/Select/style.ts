import styled from 'styled-components'

const StyledSelect = styled.select`
  appearance: none;
  width: 100%;
  height: 24px;
  background-image: url('images/arrow_up_down.svg');
  background-repeat: no-repeat;
  background-position-x: 99%;
  background-position-y: 50%;
  background-size: 10px;
  border-radius: 3px;
  padding-left: 12px;
  font-size: 14px;
  height: 2.3em;
  &:focus {
    border-color: #f5a623;
    outline: 0;
  }
`

export default StyledSelect
