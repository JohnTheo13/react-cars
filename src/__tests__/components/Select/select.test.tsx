import React from 'react'
import { render } from '@testing-library/react'
import Select from '../../../compenents/Select'


describe('Select input tests', () => {
  const props = {
    placeholder: 'placeholder',
    options: ['hello', 'world', 'of', 'cars'],
    name: 'test',
    onChange: jest.fn()
  }
  const { container, getByText, ...utils } = render(<Select {...props} />)
  test('renders the right placeholder', () => {
    expect(getByText(props.placeholder)).toBeInTheDocument()
  })
  test('Match snapshot', () => {
    expect(container).toMatchSnapshot()
  })
  // fire events and more
})
