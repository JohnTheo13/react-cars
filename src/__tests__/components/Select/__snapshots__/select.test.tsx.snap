// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select input tests Match snapshot 1`] = `<div />`;
